src/cmd/linuxkit/pkglib/pkglib.go

```diff
@@ -169,6 +172,43 @@ func NewFromCLI(fs *flag.FlagSet, args ...string) (Pkg, error) {
 		}
 	})
 
+	var srcHashes string
+	var sources []string
+	if len(pi.Sources) != 0 {
+		// If we have external sources ensure the package path is included first
+		sources = []string{pkgPath + ":/"}
+	}
+	for _, source := range pi.Sources {
+		tmp := strings.Split(source, ":")
+		if len(tmp) != 2 {
+			return Pkg{}, fmt.Errorf("Bad source format in %s", source)
+		}
+
+		// Make sure the srcPath
+		srcPath := tmp[0]
+		if !filepath.IsAbs(srcPath) {
+			srcPath, err = filepath.Abs(filepath.Join(pkgPath, srcPath))
```

Comment:
`filepath.Join(pkgPath, srcPath)` is already absolute (and `.Clean`), so `filepath.Abs` here does nothing.